core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -1796,6 +1799,48 @@ public void testCatalogExpiredBearerTokenIsRefreshedWithCredential(String oauth2
             any());
   }
 
+
+  @Test
+  public void testCatalogWithPagaintionTokenIssue() {
+    //TODO remove this test, Used to highlight issue with namespaces
```

Comment:
I don't think we should remove this test. We actually want a test that makes sure we get the expected behavior

Reply:
alternatively you can also add the mockito verifications on `adapter` to the other tests that you added (`testPaginationForListNamespaces()` / `testPaginationForListTables()` / `testPaginationForListViews()`)

Reply:
Will fix this, either by modifying the other tests i created or create a new test

Reply:
@nastra Have made the revisions to use mockito within 
```
(testPaginationForListNamespaces() / testPaginationForListTables() / testPaginationForListViews()
```